src/ipc/ipc-zephyr.c

```diff
@@ -133,13 +133,24 @@ enum task_state ipc_platform_do_cmd(struct ipc *ipc)
 	/* perform command */
 	ipc_cmd(hdr);
 
-#if CONFIG_SOC_SERIES_INTEL_ADSP_CAVS
-	/* are we about to enter D3 ? */
-	if (ipc->pm_prepare_D3) {
-		/* no return - memory will be powered off and IPC sent */
+	if (ipc->task_mask & IPC_TASK_POWERDOWN) {
+#if defined(CONFIG_PM)
+		/**
+		 * @note For primary core this function
+		 * will only force set lower power state
+		 * in power management settings.
+		 * Core will enter D3 state after exit
+		 * IPC thread during idle.
+		 */
+		cpu_disable_core(PLATFORM_PRIMARY_CORE_ID);
+#else
+		/**
+		 * @note no return - memory will be
+		 * powered off and IPC sent.
+		 */
```

Comment:
Because for zephyr platforms we are supposed to use IPC_MASK.
I've dropped commit with refactor marking prepare_D3 flag as obsolete when compiling with __ZEPHYR__
because I thought it's the reason of TGL tests failing. Turns out it's not...

Reply:
I think we're not ready to use the Zephyr APIs for PM on TGL yet, so that check would still be required.

Reply:
and oh BTW CONFIG_PM will also be set for TGL, it doesn strcitly mean using the zephyr PM right? so in fact for TGL, you'd never get to the #else part right?

Reply:
Leaving pm_prepare_D3 seems good, but might be easier to not do it in this patch. Will be easier to review that TASK_POWERDOWN cover all scenarios in a separate patch to deprecate the old flag.